drivers/spi/spi_ambiq_spis.c

```diff
@@ -0,0 +1,381 @@
+		} else {
+			data->ios_cfg.ui32InterfaceSelect |= AM_HAL_IOS_SPIMODE_0;
+		}
+	}
+
+	if (config->operation & SPI_OP_MODE_MASTER) {
+		LOG_ERR("Controller mode not supported");
+		return -ENOTSUP;
+	}
+	if (config->operation & SPI_MODE_LOOP) {
+		LOG_ERR("Loopback mode not supported");
+		return -ENOTSUP;
+	}
+
+	if (spi_cs_is_gpio(config)) {
+		LOG_ERR("CS control via GPIO is not supported");
+		return -EINVAL;
+	}
+
+	/* Eliminate the "read-only" section, so an external host can use the
+	 * entire "direct write" section.
+	 */
+	data->ios_cfg.ui32ROBase = FIFO_BASE;
+	/* Making the "FIFO" section as big as possible. */
+	data->ios_cfg.ui32FIFOBase = FIFO_BASE;
+	/* We don't need any RAM space, so extend the FIFO all the way to the end
+	 * of the LRAM.
+	 */
+	data->ios_cfg.ui32RAMBase = FIFO_END;
+	/* FIFO Threshold - set to half the size */
+	data->ios_cfg.ui32FIFOThreshold = FIFO_LENGTH >> 1;
+
+	data->ios_cfg.pui8SRAMBuffer = g_pui8TxFifoBuffer,
+	data->ios_cfg.ui32SRAMBufferCap = AMBIQ_SPIS_TX_BUFSIZE_MAX,
+
+	ctx->config = config;
+
+	ret = am_hal_ios_configure(data->ios_handler, &data->ios_cfg);
+
+	return ret;
+}
+
+static int spi_ambiq_xfer(const struct device *dev, const struct spi_config *config)
+{
+	const struct spi_ambiq_config *cfg = dev->config;
+	struct spi_ambiq_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	int ret = 0;
+	uint32_t chunk, num_written, num_read = 0;
+
+	while (1) {
+		if (spi_context_tx_buf_on(ctx)) {
+			/* Inform the host */
+			gpio_pin_set_dt(&cfg->int_gpios, 1);
+			k_sleep(K_USEC(100));
+			gpio_pin_set_dt(&cfg->int_gpios, 0);
+			/*  Copy data into FIFO */
+			chunk = (ctx->tx_len > AMBIQ_SPIS_TX_BUFSIZE_MAX)
+					? AMBIQ_SPIS_TX_BUFSIZE_MAX
+					: ctx->tx_len;
+			ret = am_hal_ios_fifo_write(data->ios_handler, (uint8_t *)ctx->tx_buf,
+						    chunk, &num_written);
+			if (ret != 0) {
+				spi_ambiq_reset(dev);
+				LOG_ERR("SPIS write error: %d", ret);
+				return ret;
+			}
+			spi_context_update_tx(ctx, 1, num_written);
+			if (!spi_context_tx_buf_on(ctx)) {
+				break;
+			}
+		} else if (spi_context_rx_buf_on(ctx)) {
+			/* Wait for host write complete interrupt */
+			(void)k_sem_take(&data->spim_wrcmp_sem, K_FOREVER);
+			/* Read out the first byte as packet length */
+			num_read = am_hal_ios_pui8LRAM[0];
+			if (ctx->rx_len < num_read) {
+				LOG_ERR("SPIS read buffer inadequate!");
+				return -ENOMEM;
+			}
+			/* Read data from LRAM */
+			memcpy(ctx->rx_buf, (uint8_t *)&am_hal_ios_pui8LRAM[1], num_read);
+			spi_context_update_rx(ctx, 1, num_read);
+			break;
+		}
+	}
+
+	spi_context_complete(ctx, dev, ret);
+	return ret;
+}
+
+static int spi_ambiq_transceive(const struct device *dev, const struct spi_config *config,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	struct spi_ambiq_data *data = dev->data;
+	int ret;
+
+	if (!tx_bufs && !rx_bufs) {
+		return 0;
+	}
+
+	ret = pm_device_runtime_get(dev);
+
```

Comment:
remove that empty line, as you are testing ret in the if statement. apply that everywhere where relevant.